### PR TITLE
add_support_for_contract_stat

### PR DESCRIPTION
**Pull request by ElvisLYC**
I added `contract_stat` for two  futures exchanges (bybit and okex_swap) to test out two possible scenario. Please refer to my comments below.

### CODE REVIEW

File: lib/cryptoexchange/exchanges/bybit/services/contract_stat.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+module Cryptoexchange::Exchanges
```
Comment (ElvisLYC) [inline lib/cryptoexchange/exchanges/bybit/services/contract_stat.rb]: this is a scenario where `open_interest` and `index` data came from the same `url`

File: lib/cryptoexchange/exchanges/okex_swap/services/contract_stat.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+module Cryptoexchange::Exchanges
```
Comment (ElvisLYC) [inline lib/cryptoexchange/exchanges/okex_swap/services/contract_stat.rb]: this is a scenario where `open_interest` and `index` data  came from different url

File: spec/exchanges/bybit/integration/market_spec.rb [hunk 3]
```diff
@@ -52,5 +52,18 @@
     expect(order_book.bids.count).to be > 0
     expect(order_book.timestamp).to be_nil
     expect(order_book.payload).to_not be nil
-  end  
+  end
+
+  it 'fetch contract stat' do
```
Comment (ElvisLYC) [inline spec/exchanges/bybit/integration/market_spec.rb]: added spec for `contract_stat`

File: lib/cryptoexchange/exchanges/okex_swap/services/contract_stat.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        def fetch(market_pair)
+          unless market_pair.contract_interval == "perpetual"
+            open_interest = super(open_interest_url(market_pair))
+            index = super(index_url(market_pair))
+            adapt(open_interest, index, market_pair)
+          end
+        end
+
+        def open_interest_url(market_pair)
+          "https://www.okex.com/api/futures/v3/instruments/#{market_pair.inst_id}/open_interest"
```
Comment (tmlee) [inline lib/cryptoexchange/exchanges/okex_swap/services/contract_stat.rb]: If the URL domain/prefix are shared from the `market.rb`,you can use that constant

File: lib/cryptoexchange/exchanges/bybit/services/market.rb [hunk 5]
```diff
@@ -27,7 +27,7 @@ def adapt_all(output)
             market_pair = Cryptoexchange::Models::MarketPair.new(
               base: base,
               target: target,
-              market: Btcnext::Market::NAME
+              market: Bybit::Market::NAME
```
Comment (tmlee) [inline lib/cryptoexchange/exchanges/bybit/services/market.rb]: is this an unrelated mistake ?

Comment (ElvisLYC) [reply lib/cryptoexchange/exchanges/bybit/services/market.rb]: yes noticed this mistake from past implementation

File: lib/cryptoexchange/exchanges/okex_swap/services/contract_stat.rb (hunk 4 above)
Comment (ElvisLYC) [reply lib/cryptoexchange/exchanges/okex_swap/services/contract_stat.rb]: they are not the same.. anyway, i have assign them to a `CONTRACT_STAT_URL` as it is repetitive

File: lib/cryptoexchange/models/contract_stat.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  module Models
+    class ContractStat
+      attr_accessor :base, :target, :market, :open_interest,
+                    :index, :timestamp, :payload
+
+      def initialize(args = {})
+        @base = args[:base]
+        @target = args[:target]
+        @market = args[:market]
+        @open_interest = args[:open_interest] || []
```
Comment (wongy91) [inline lib/cryptoexchange/models/contract_stat.rb]: why not just return nil?

File: lib/cryptoexchange/models/contract_stat.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        @index = args[:index] || []
```
Comment (wongy91) [inline lib/cryptoexchange/models/contract_stat.rb]: same here

File: lib/cryptoexchange/models/contract_stat.rb (hunk 6 above)
Comment (ElvisLYC) [reply lib/cryptoexchange/models/contract_stat.rb]: noted and changed

**Pull request merged**